flink-java/src/main/java/org/apache/flink/api/java/io/CsvInputFormat.java

```diff
@@ -235,8 +252,21 @@ public OUT readRecord(OUT reuse, byte[] bytes, int offset, int numBytes) throws
 		
 		if (parseRecord(parsedValues, bytes, offset, numBytes)) {
 			// valid parse, map values into pact record
-			for (int i = 0; i < parsedValues.length; i++) {
-				reuse.setField(parsedValues[i], i);
+			if (pojoTypeInfo == null) {
+				Tuple result = (Tuple) reuse;
+				for (int i = 0; i < parsedValues.length; i++) {
+					result.setField(parsedValues[i], i);
+				}
+			} else {
+				for (int i = 0; i < parsedValues.length; i++) {
+					try {
+						pojoTypeInfo.getPojoFieldAt(i).field.set(reuse, parsedValues[i]);
```

Comment:
I'm not 100% sure if this also works for pojo's with private fields.
In your testcase, all the fields are public. I'm not sure where exactly we are setting the fields accessible, but it might be the case that they are not set accessible here. 
Can you add a private field with a getter/setter to the test pojo to validate this?